samples/matter/lock/src/app_task.cpp

```diff
@@ -274,6 +288,71 @@ void AppTask::LockActionEventHandler(const AppEvent &event)
 	}
 }
 
+#ifdef CONFIG_THREAD_WIFI_SWITCHING
+void AppTask::SwitchImagesEventHandler(const AppEvent &event)
+{
+	/* Swap the application from using Thread to Wi-Fi or opposite. */
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+	LOG_INF("Switching application from Thread to Wi-FI triggered.");
+	SystemLayer().ScheduleLambda([] {
+		SoftwareImagesSwapper swapper;
+		swapper.Swap(SoftwareImagesSwapper::ApplicationImageId::Image_2);
+	});
+#elif defined(CONFIG_CHIP_WIFI)
+	LOG_INF("Switching application from Wi-Fi to Thread triggered.");
+	SystemLayer().ScheduleLambda([] {
+		SoftwareImagesSwapper swapper;
+		swapper.Swap(SoftwareImagesSwapper::ApplicationImageId::Image_1);
+	});
+#endif
+}
+
+void AppTask::SwitchImagesTimerTimeoutCallback(k_timer *timer)
+{
+	if (!timer) {
+		return;
+	}
+
+	Instance().mSwitchModesTimerActive = false;
+
+	AppEvent event;
+	event.Type = AppEventType::Timer;
+	event.TimerEvent.Context = k_timer_user_data_get(timer);
```

Comment:
Do we ever use this context or `k_timer_user_data_get`? Looks we set there the pointer to the application, but it's global, so I'm not sure it's useful at all.